### DISCUSSION TITLE
Triggering GitHub-hosted runners without pushing to the remote repo

### DISCUSSION THREAD

**Discussion by sudg6** [community/community#132648] [Actions] [answered]
### Select Topic Area

Question

### Body

Knowing that the `workflow.yml` script is unchanged all the time, I am making frequent changes to local text files from my local repo and the expected output from the building process defined in the `workflow.yml` is image figures (.png files) by performing calculations, however, not each change makes sens to be committed and pushed to the remote repo.

Is it possible anyhow for the local script `work.yml` instead of the remote one, (or if possible any other local script should be created) to trigger the runner by reading from the local text files without pushing them altogether to the remote repo as usual?
If this is possible, the virtual machine (runner) would read from local changes and make the calculations happen in the cloud and returning the png files without affecting the remote repository?

**Comment by knarrff** [comment, marked as answer]
I do not think this is possible: you would have to push things to github for runners to see and that has to be through the repo.

What I would suggest is that for your current work (that you don't necessarily want to end up in your main branch), setup another branch and use that. Do commit and push there and get the runners to run there as usual. Once you are satisfied with the result, you can squash-merge the work into your main branch (so that all the intermediate commits will be gone and all work from that branch will look like a single commit in the end, without intermediate steps. Beware: do not squash-merge when there are others working on that branch as well.

**Reply by knarrff** [reply]
Of course, if you can locally run whatever it is the runners run, this would usually be the preferred method. But this would not involve the runners at all: your local computer would do the work. This usually requires some setup, but in the end often is faster and has no dependency on a network connection, no queues, ...

**Reply by sudg6** [reply]
Absolutely!
Interacting locally with Github machines  currently is not possible. The only way to utilize the resources is through submitting jobs via git which exclusively implies the remote repository to be up-to-date with the local one.
